src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> last;
+    private Node<T> first;
+    
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E element, Node<E> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+    
+    void linkLast(T value) {
+        Node<T> lastNode = last;
+        Node<T> newNode = new Node<>(lastNode, value, null);
+        last = newNode;
+        if (lastNode == null) {
+            first = newNode;
+        } else {
+            lastNode.next = newNode;
+        }
+        size++;
+    }
+    
+    void linkBefore(T value, Node<T> succ) {
+        Node<T> pred = succ.prev;
+        Node<T> newNode = new Node<>(pred, value, succ);
+        succ.prev = newNode;
+        if (pred == null) {
+            first = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+    }
+    
     @Override
     public void add(T value) {
+        linkLast(value);
     }
-
+    
     @Override
     public void add(T value, int index) {
+        if (indexCheckAdd(index)) {
+            if (index == size) {
+                linkLast(value);
+            } else {
+                linkBefore(value, findNodeByIndex(index));
+            }
+        } else {
+            throw new IndexOutOfBoundsException("Out of bounds exception for"
+                        + " add(value, index) method");
+        }
+    }
+    
+    private boolean indexCheckAdd(int index) {
+        return index >= 0 && index <= size;
+    }
+    
+    private boolean indexCheckSetGetRemove(int index) {
+        return index >= 0 && index < size;
+    }
+    
+    Node<T> findNodeByIndex(int index) {
+        if (index < size / 2) {
+            Node<T> foundNode = first;
+            for (int i = 0; i < index; i++) {
+                foundNode = foundNode.next;
+            }
+            return foundNode;
+        } else {
+            Node<T> foundNode = last;
+            for (int i = size - 1; i > index; i--) {
+                foundNode = foundNode.prev;
+            }
+            return foundNode;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (indexCheckSetGetRemove(index)) {
```

Comment:
same here) let's remove redundant `else` block and throw exception in `if` block